### ISSUE TITLE
[Feature] Show pages continuously, 4 pages a time, but use scrollwheel for navigation.

### ISSUE THREAD

**Issue by enricorov** [sumatrapdfreader/sumatrapdf#1171]
I often find myself using this program for coursework, which involves great amounts of pdf-converted packs of slides, like so:
![immagine](https://user-images.githubusercontent.com/23199149/58867179-45869e00-86ba-11e9-8652-afcce0334ca5.png)

I would like to now move backwards or forwards by two slides like I would with the spacebar, I get this:
![immagine](https://user-images.githubusercontent.com/23199149/58867413-aca45280-86ba-11e9-8f71-223501ecf62a.png)

Which is not quite usable compared to the effect a SHIFT + Space press, which does this:
![immagine](https://user-images.githubusercontent.com/23199149/58867557-f856fc00-86ba-11e9-876e-063aaeef93e3.png)

I would suggest splitting "Show pages continuously" into "Display pages continuously" and "Scroll pages continuously", so that deactivating it doesn't lead to this:
![immagine](https://user-images.githubusercontent.com/23199149/58867832-72878080-86bb-11e9-8483-ebf5ae1d2186.png)

I find this to be the only thing missing from perfect mouse navigation. I love this feature, and would like to use it all the time. Thanks in advance, and congrats on the program. It's super lightweight and usable.

**Comment by GitHubRulesOK** [comment]
Your Page layout is slightly less conventional since they appear to be  more like slide or flashcard layouts
The Landscape aspect naturally allows 4 up rather than the more usual 2pages showing and the desire to show 3 or 9 pages etc at a time is a separate issue. (you can at least see 4 by using zoom).
The controls for jumping 4 pages would probably be needed if that feature were to be ever written, so I will reference link this request there. 

Currently, depending on settings, you probably need Ctrl Left arrow (previous page pair) and Ctrl Right arrow (next page pair). However you ask for that to be driven by mouse scroll which is a "windows" control that SumartaPDF will respond to but has no means to change. In windows settings there are rudimentary ways to tell the wheel to scroll pages rather than lines BUT I have not usually found it worth changing as it then often messes up other usage.  (Microsoft sometimes charged for better mouse drivers)

There are 3rd part apps that allow you to change mouse (and keyboard) behaviour per application and I personally would be using something like Hirez XMouse Button Control (XMBC) http://www.highrez.co.uk/downloads/XMouseButtonControl.htm to set the mouse functions to emulate ctrl arrows

**Comment by enricorov** [comment]
> you ask for that to be driven by mouse scroll which is a "windows" control that SumartaPDF will respond to but has no means to change. In windows settings there are rudimentary ways to tell the wheel to scroll pages rather than lines BUT I have not usually found it worth changing as it then often messes up other usage.

Yes, I tried doing that, and the behaviour would conflict in more conventional layouts. I'd suggest making this a "fullscreen" or slideshow mode, since this basically is it, and then have the mouse scroll differently there.
I'm already using AHK, do you know if I can map the wheel to the arrow keys with it?

**Comment by GitHubRulesOK** [comment]
Sorry although I use AHK occasionally for other reasons, I have no knowledge of "steering wheels" :-) but seem to recollect others may have.

**Comment by enricorov** [comment]
Thanks a lot, I'll look around.

**Comment by SumatraPeter** [comment]
> Currently, depending on settings, you probably need Ctrl Left arrow (previous page pair) and Ctrl Right arrow (next page pair). However you ask for that to be driven by _mouse scroll which is a "windows" control that SumartaPDF will respond to but has no means to change_.

Technically I see no reason that this can't be done. Imagine you're developing an FPS game for example. You are completely free to make the mousewheel up/down scroll events result in different weapons being selected. Windows doesn't force every single program to respond to mouse (and other) events only in fixed ways - that would be ridiculous and very limiting. Thus as far as I can tell nothing stops a dev from coding Sumatra to respond to mousewheel scroll events with pairwise page movement, especially when certain page display conditions are met. Only million dollar question is, is this something that will benefit many and if so, is there anyone who will care to write the required code?

**Comment by enricorov** [comment]
> Thus as far as I can tell nothing stops a dev from coding Sumatra to respond to mousewheel scroll events with pairwise page movement, especially when certain page display conditions are met.
That's what I thought, but I guess it's quite a niche issue.

**Comment by nod5** [comment]
> I'm already using AHK, do you know if I can map the wheel to the arrow keys with it?

https://www.autohotkey.com/docs/KeyList.htm#mouse-wheel

Minimal AutoHotkey script to use mouse wheel to trigger `Ctrl+Left` and `Ctrl+Right` shortcuts when SumatraPDF is active.
```
#IfWinActive ahk_exe SumatraPDF.exe
WheelUp::   Send ^{Left}
WheelDown:: Send ^{Right}
#IfWinActive
```

Less minimal AutoHotkey script that uses `F10` to toggle the mouse scroll whole page mode on/off.

```
#IfWinActive ahk_exe SumatraPDF.exe
F10:: 
wheel_mode := wheel_mode = "on" ? "off" : "on"
tooltip % "Mouse Scroll Whole Page: " wheel_mode
Hotkey, IfWinActive, ahk_exe SumatraPDF.exe
Hotkey, WheelUp, wheel_up, %wheel_mode%
Hotkey, WheelDown, wheel_down, %wheel_mode%
Hotkey, IfWinActive
sleep 800
tooltip
return

wheel_up:
Send ^{Left}
return

wheel_down:
Send ^{Right}
return
#IfWinActive
```

**Comment by GitHubRulesOK** [comment]
Hi Nod5 I should have remembered you have much experience in writing helper scripts
I love the first minimalist example and using a hot key is great though F10 is a duplicate I would suggest F7 see https://github.com/GitHubRulesOK/MyNotes/raw/master/AppNotes/SumatraPDF/Keys%20and%20Combinations.pdf

**Comment by enricorov** [comment]
Thanks. I'm now using the AHK script with F7 to toggle. I'd leave the issue open, since this is a workaround.

**Comment by GitHubRulesOK** [comment]
@kjk this request is partly met by setting scroll to whole pages "Mouse Wheel Turns Page" will in this case roll a pair when side by side.
However the number of pages is not settable like some other commands. (Though one could argue 2 notches meets the OP desire)
Thus similar to step pages by N this command could possibly be extended a similar way to 2 or 4 as required?

<img width="1551" height="644" alt="Image" src="https://github.com/user-attachments/assets/32b21f3e-eb3c-49e5-937d-b9f8e30ebbd5" />
